meson_options.txt

```diff
@@ -1,3 +1,4 @@
 option('usewebcam', type: 'boolean', value: false, description: 'Build with webcam support (opencv)')
 option('usemididrum', type: 'boolean', value: false, description: 'Build with midi drum support (portmidi)')
 option('usewebserver', type: 'boolean', value: false, description: 'Build with web server support (cpprest)')
+option('stoponwarning', type: 'boolean', value: false, description: 'Stop compilation if a warning is detected')
```

Comment:
To be fair i wouldn't make this into an option. The code should be warning free imho. Blocking it at the PR level can be seen as 'annoying' since they 'didn't expect that'

I'd suggest to always use the beforementioned arguments. What are your thoughts on this @deuteragenie ?

Reply:
I think this is too hard and unpredictable, which is the reason I made it an option.
For example, I compiled it under gcc 8 and 11, but not on other versions.  Maybe these issue warnings ? Or maybe later gcc versions ?
So by turning it on on CI, we ensure that no warnings creep in ON ci.
However, if/when local devs or packagers want to compile it, this will be by default off.  They can (and should) turn it on if they wish.